src/contracts/ops/DelegateDeployer.sol

```diff
@@ -0,0 +1,30 @@
+pragma solidity 0.6.2;
+pragma experimental ABIEncoderV2;
+
+// A delegate deployer that reverts upon failure
+// and emits the address if successful.
+// @author Patrick McCorry
+contract DelegateDeployer {
+
+    event Deployed(address _contract);
+
+    /// Note: This function is intended for DELEGATECALL.
+    /// If you use CALL instead of DELEGATECALL, then:
+    /// - constructor cannot use msg.sender (pass owner as constructor or use init())
+    /// - ensure msg.value == _value.
+    /// @dev Deploys a contract via CREATE2.
+    /// @param _initCode Bytecode for deployment
+    /// @param _value Specify coins to deploy with.
+    /// @param _salt Hash of extra data.
+    function deploy(bytes memory _initCode, uint _value, bytes32 _salt) public payable
```

Comment:
I think this should return the address - could be useful if you want to use the contract you've just deployed without having to calc the address.

If you do that you wont need to declare the addr below, since you can just declare it in the return

Reply:
you mean the event? i've just removed the event. since we are using create2, its deterministic anyway.